### DISCUSSION TITLE
Import historical data?

### DISCUSSION THREAD

**Discussion by intelfx** [netdata/netdata#15177] [Q&A]
I just wrote a custom netdata plugin to collect certain hardware metrics. Now, I have a bunch of historical data (in format of this plugin) that I would like to import to Netdata to visualize.

What would be the best way to do so?

(I understand I'll have to get hacking on the Netdata agent itself. I'm merely asking for pointers on where to get started and what would be the easiest way to plumb that through the code.)

**Comment by andrewm4894** [comment]
interesting question @intelfx - i am asking the @netdata/agent team about this. 

if we dont have anything then i for sure think should be a feature request we should make.

**Comment by MrZammler** [comment]
Hi @intelfx . Indeed we don't have something straight forward to do this (or I'm not aware of it myself). One pointer I could give you is to check the replication code. Essentially what it does is use similar keywords as the plugins do, but also with the ability to back fill old metrics on a chart.

Do let us know if you need any assistance.

**Reply by intelfx** [reply]
I see, thanks.

Can I just change the plugin in question to emit "replication" keywords containing the backfill instructions instead of normal BEGIN/CHART/DIMENSION/SET, or are these keywords limited to the replication data path only?

**Reply by andrewm4894** [reply]
fyi - i think this is the agent code that does streaming and replication: https://github.com/netdata/netdata/tree/master/streaming

**Comment by andrewm4894** [comment]
ps @intelfx if your collector might be useful to others it would be cool to add it in here: https://github.com/netdata/community/tree/main/collectors

(i can help)

**Reply by intelfx** [reply]
Yep, I was planning to do that once I polish the code enough and add a couple of non-essential features.
